### PR TITLE
Setup engines tests

### PR DESCRIPTION
**Pull request by sauloperez**
2016-08-19-163318_1676x1007_scrot

### CODE REVIEW

**Comment by sauloperez** [comment]
This is a *trenecito* of https://github.com/coopdevs/katuma/pull/76

File: .travis.yml [hunk 1]
```diff
@@ -1,11 +1,14 @@
...
 
+env:
+  - RAILS_ENV=test
+
+before_install:
+  - gem update bundler
+
 before_script:
-  - "bundle exec rake db:setup"
+  - psql -c 'create database katuma_test;' -U postgres
```
Comment (enricostano) [inline .travis.yml]: ditto in other PR

**Comment by enricostano** [comment]
> I couldn't run the specs because of this association. The test database doesn't have any users table so we'll need to run somehow the migrations of the engines this one depends on.

Ouch!

File: Rakefile [hunk 2]
```diff
@@ -2,6 +2,12 @@
 # Add your own tasks in files placed in lib/tasks ending in .rake,
 # for example lib/tasks/capistrano.rake, and they will automatically be available to Rake.
 
+require 'byebug'
```
Comment (enricostano) [inline Rakefile]: :eyes:

File: engines/group/spec/dummy/db/schema.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+# you'll amass, the slower it'll run and the greater likelihood for issues).
+#
+# It's strongly recommended that you check this file into your version control system.
+
+ActiveRecord::Schema.define(version: 20150328134731) do
+
+  # These are extensions that must be enabled in order to support this database
+  enable_extension "plpgsql"
+
+  create_table "groups", force: :cascade do |t|
```
Comment (enricostano) [inline engines/group/spec/dummy/db/schema.rb]: `force: :cascade` explain please :pray:

Comment (enricostano) [inline engines/group/spec/dummy/db/schema.rb]: I mean... not `EXPLAIN`... lol

Comment (sauloperez) [inline engines/group/spec/dummy/db/schema.rb]: lol, I don't know... Ask my rails, this is autogenerated...

Comment (enricostano) [inline engines/group/spec/dummy/db/schema.rb]: http://stackoverflow.com/questions/27697119/what-is-cascade-in-rails-schema-rb-and-where-did-it-come-from

Comment (sauloperez) [inline engines/group/spec/dummy/db/schema.rb]: Now we know. It makes sense

**Comment by sauloperez** [comment]
After some more investigation, I'm seriously considering no to test with dummy apps in each engine.

Turns out that the engine has a `katuma_<engine_name>_test` database where you first run your migrations. This is fine when the engine has little dependencies and does not depend on another engine's migrations. However, when that's not the case, things get though. 

Group's engine depends on Account's engine to set up the users table - it's the engine that has write access to it. If Account was properly set up with its dummy app (not yet done) it'd have its own `katuma_account_test` database. So, for Group to work, they'd need to share a database, which in the end, is just testing without dummy app.

That way you would run all the engines' migrations before running Group's engine tests. That means that all engines need to see all other engines' migrations.

**Comment by enricostano** [comment]
Another place to look for some answer aside the [CBRA book](https://leanpub.com/cbra) is https://groups.google.com/forum/#!forum/components-in-rails

File: engines/shared/script/test.sh [hunk 4]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+exit_code=0
+engine_name=`pwd | cut -f6 -d '/'`
```
Comment (Morantron) [inline engines/shared/script/test.sh]: `engine_name=$(basename $PWD)`

File: script/test.sh [hunk 5]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+result=0
+
+for engine_name in $(ls -d engines/*/ | cut -f 2 -d '/'); do
```
Comment (Morantron) [inline script/test.sh]:
```bash
for engine_script in engines/*/script/test.sh; do
   pushd $(dirname $engine_script)
      $(./$(basename $engine_script)
   popd
done
```

Comment (Morantron) [inline script/test.sh]: maybe it doesn't work lol :trolleybus:

**Comment by sauloperez** [comment]
Q ase este tio aqui!

**Pull request merged**